### ISSUE TITLE
print doesn't change the line

### ISSUE THREAD

**Issue by David-UPCT** [wxMaxima-developers/wxmaxima#1240]
Hi.

I've just updated wxmaxima to 20.03.1 in Archlinux, I don't know if this is a bug or a change in the behaviour, but in this version the **print** function doesn't change the line. Please see the images I've attached.

![19 09 0](https://user-images.githubusercontent.com/5481139/78893304-6bfc2f00-7a6b-11ea-8d37-d2c12023399d.png)
![20 03 1](https://user-images.githubusercontent.com/5481139/78893319-71f21000-7a6b-11ea-88a5-5c67d1caafaf.png)

**Comment by David-UPCT** [comment]
Update: similar behaviour with **disp** and **display**. It's different to the behaviour of maxima in a terminal.
![Captura de pantalla de 2020-04-21 19-39-24](https://user-images.githubusercontent.com/5481139/79895977-06635780-8408-11ea-9bba-8d70ceb20817.png)

**Comment by KubaO** [comment]
This is a recent regression. I'll have to bisect it - I've noticed this behavior as well.

**Comment by gunterkoenigsmann** [comment]
The good news is that I can reproduce it on my system. The bad news is that the code involved is way too complicated. Perhaps I should first simplify it and then fix that bug hoping that this approach doesn't introduce new bugs.

**Comment by gunterkoenigsmann** [comment]
@KubaO: Thanks a lot for volunteering for bisecting this one: It is always good to have someone who is willing to do this kind of work. But I believe I have resolved this bug now.

@all: As whilst trying to resolve this bug I have strongly simplified code that I *hope* I understood I cannot guarantee that you won't find new bugs I have created right now, though.
Sometimes improving things is an iterative process.
